tool/tsh/common/kube_proxy.go

```diff
@@ -497,6 +520,39 @@ func issueKubeCert(ctx context.Context, tc *client.TeleportClient, proxy *client
 	return cert, nil
 }
 
+// checkMultipleClusterSelections takes a map of name selectors to matched
+// clusters and checks that each matching is valid.
+func checkMultipleClusterSelections(cf *CLIConf, tc *client.TeleportClient, matchMap map[string]types.KubeClusters) error {
+	for name, clusters := range matchMap {
+		err := checkClusterSelection(cf, tc, clusters, name)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	}
+	return nil
+}
+
+// combineMatchedClusters combineMatchedClusters takes a map from name selector
+// to matched clusters and combines all the matched clusters into a deduplicated
+// slice.
+func combineMatchedClusters(cf *CLIConf, tc *client.TeleportClient, matchMap map[string]types.KubeClusters) types.KubeClusters {
+	var out types.KubeClusters
+	for _, clusters := range matchMap {
+		out = append(out, clusters...)
+	}
+	return types.DeduplicateKubeClusters(out)
+}
+
+// matchClustersByNames maps each name to the clusters it matches by exact name
+// or by prefix.
+func matchClustersByNames(cf *CLIConf, tc *client.TeleportClient, clusters types.KubeClusters, names ...string) map[string]types.KubeClusters {
```

Comment:
`cf` param is unused: 
```suggestion
func matchClustersByNames(tc *client.TeleportClient, clusters types.KubeClusters, names ...string) map[string]types.KubeClusters {
```